### ISSUE TITLE
Ways to reduce compile time

### ISSUE THREAD

**Issue by Nutomic** [LemmyNet/lemmy#810]
- [x] remove brotli dependency from actix ([issue link](https://github.com/actix/actix-web/issues/1561))
- [x] create seperate cargo workspaces (#809)
- [ ] wait for lettre 0.10 which makes the openssl dependency optional
- [x] figure out how to use lld in docker builds
- [x] remove unnecessary macro definitions in db

**Comment by Nutomic** [comment]
This is mostly finished, except lettre 0.10 which we will use once its ready for production.

**Comment by robjtede** [comment]
Out of interest, how much of a difference have these efforts made?

**Comment by Nutomic** [comment]
- lld reduced linking time from 10s to 5s
- removing unused macros didnt change anything
- removing brotli might save 30s of a single core (but probably way less if you have many cores)
- workspaces didnt change anything because none of them can be compiled in parallel (but it helped with better organising the code)

In fact our compilation time has actually gone up a lot due to new dependencies and new code we added. Maybe `cargo-bloat` or other tools could help to speed up compilation, but I dont understand how they work and this is a really low priority for us.
